src/drawing-tool/components/app.tsx

```diff
@@ -0,0 +1,216 @@
+
+interface StampCollection {
+  collection: "molecules" | "ngsaObjects" | "custom";
+  name?: string;
+  stamps?: string[];
+}
+
+export interface IAuthoredState extends IAuthoringOpenResponseMetadata {
+  version: number;
+  hint?: string;
+  backgroundImageUrl?: string;
+  imageFit: string;
+  imagePosition: string;
+  stampCollections: StampCollection[];
+}
+
+export interface IInteractiveState extends IRuntimeInteractiveMetadata {}
+
+const baseAuthoringProps = {
+  schema: {
+    type: "object",
+    definitions: {
+      stampCollection: {
+        title: "Stamp collection",
+        type: "object",
+        properties: {
+          collection: {
+            type: "string",
+            enum: [
+              "molecules",
+              "ngsaObjects",
+              "custom"
+            ],
+            enumNames: [
+              "Molecules (predefined)",
+              "NGSA Objects (predefined)",
+              "Custom"
+            ],
+            default: "molecules"
+          }
+        },
+        required: [
+          "collection"
+        ],
+        dependencies: {
+          collection: {
+            oneOf: [
+              {
+                properties: {
+                  collection: {
+                    enum: [
+                      "molecules"
+                    ]
+                  },
+                  name: {
+                    title: "Collection title",
+                    type: "string",
+                    default: "Molecules"
+                  }
+                }
+              },
+              {
+                properties: {
+                  collection: {
+                    enum: [
+                      "ngsaObjects"
+                    ]
+                  },
+                  name: {
+                    title: "Collection title",
+                    type: "string",
+                    default: "Objects"
+                  }
+                }
+              },
+              {
+                properties: {
+                  collection: {
+                    enum: [
+                      "custom"
+                    ]
+                  },
+                  name: {
+                    title: "Collection title",
+                    type: "string",
+                    default: "My Stamps"
+                  },
+                  stamps: {
+                    title: "Stamps",
+                    type: "array",
+                    items: {
+                      type: "string",
+                      default: "https://"
+                    },
+                    default: ["https://"]
+                  }
+                },
+                required: [
+                  "name"
+                ]
+              }
+            ]
+          }
+        }
+      },
+    },
+    properties: {
+      version: {
+        type: "number",
+        default: 1
+      },
+      questionType: {
+        type: "string",
+        default: "drawing_tool"
```

Comment:
I don't think it's a valid type. I guess you should use "iframe_interactive" as its the default one. This is important for reporting. If you used "open_response" or "multiple_choice", LARA and Teacher Report would expect certain fields and they would report this question as open_response/multiple_choice, etc.

Reply:
👍